_rules/document-has-main-landmark-b40fd1.md

```diff
@@ -0,0 +1,194 @@
+---
+id: b40fd1
+name: HTML page has a main landmark
+rule_type: atomic
+description: |
+  This rule checks that each page has an element with a semantic role of `main`
+accessibility_requirements:
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+  assets:
+    - _Romance of the Three Kingdoms_ by Luo Guanzhong, translation by Charles Henry Brewitt-Taylor (Tuttle Publishing, 1925, ISBN 9780804834674)
+    - _Three Kingdoms_ by Luo Guanzhong, translation by Moss Roberts (Foreign Language Press, 1976, ISBN 7-119-00590-1)
+    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectations
+
+Within the test target, there is at least one element for which all the following are true:
+
+- the element has [semantic role][] of `main`; and
+- the element is [included in the accessibility tree][].
+
+## Assumptions
+
+- This rule assumes that the `main` [landmark][] is correctly used to identify the [main block of content][] of the page.
```

Comment:
I'm not so sure this should be an assumption. It's a rather massive one.

Reply:
I think this is OK for 2 reasons:
* I am not sure it is that big. As in, do you often see `main` elements that are used for something else than the main content?
* Ultimately, if the assumption does not hold, this will let the rule pass too often (by accepting a page with a `main` landmark even though it is used for something else and has no way to bypass blocks). So we're only generating false negatives, not false positives, and I'm fairly OK with that.